src/sharing-service.ts

```diff
@@ -292,14 +301,21 @@ export class SharingService {
     const endpoint = new URL('notebooks', this.api_url);
 
     const token = await this.token;
+    const body = JSON.stringify(requestData);
     const response = await fetch(endpoint, {
       method: 'POST',
       headers: await this.makeHeaders(token),
-      body: JSON.stringify(requestData)
+      body
     });
 
     if (!response.ok) {
-      throw new Error(`Sharing notebook failed: ${response.statusText}`);
+      const sizeInMB = new Blob([body]).size / (1000 * 1000);
+      if (sizeInMB > MAX_NOTEBOOK_SIZE_MB || response.status === CONTENT_TOO_LARGE) {
+        throw new Error(
+          `The notebook appears to be too large for the backend to handle (${sizeInMB.toFixed(2)} MB). Details: ${response.status} ${response.statusText}`
+        );
+      }
+      throw new Error(`Server share request failed: ${response.status} ${response.statusText}`);
```

Comment:
Should we say in this sentence that the limit is 10 MB? It'd be great if we could mention how much over they are compared to the size limit. Sorry for the back and forth; I know I didn’t mention it in my last review.

Reply:
So I am on the fence here: I did not want to include the specific limit in error message because the frontend does not know what the limit is configured for a given deployment on the backend.